crates/light-client-prover/src/tests/mod.rs

```diff
@@ -0,0 +1,339 @@
+mod test_utils;
+
+use sov_mock_da::{MockBlockHeader, MockDaSpec, MockDaVerifier};
+use sov_mock_zkvm::MockZkGuest;
+use sov_rollup_interface::zk::LightClientCircuitInput;
+use test_utils::{create_mock_blob, create_prev_lcp_serialized};
+
+use crate::circuit::{run_circuit, LightClientVerificationError};
+
+#[test]
+fn test_light_client_circuit_valid_da_valid_data() {
+    let light_client_proof_method_id = [1u32; 8];
+    let batch_proof_method_id = [1u32; 8];
+    let da_verifier = MockDaVerifier {};
+
+    let blob_1 = create_mock_blob([1u8; 32], [2u8; 32], 2, true);
+    let blob_2 = create_mock_blob([2u8; 32], [3u8; 32], 3, true);
+
+    let block_header_1 = MockBlockHeader::from_height(1);
+
+    let input = LightClientCircuitInput::<MockDaSpec> {
+        previous_light_client_proof_journal: None,
+        light_client_proof_method_id,
+        da_block_header: block_header_1,
+        da_data: vec![blob_1, blob_2],
+        inclusion_proof: [1u8; 32],
+        completeness_proof: (),
+        l2_genesis_state_root: Some([1u8; 32]),
+        batch_proof_method_id,
+        batch_prover_da_pub_key: [9; 32].to_vec(),
+    };
+
+    let serialized_input = borsh::to_vec(&input).expect("should serialize");
+
+    let mut guest = MockZkGuest::new(serialized_input);
+
+    let output_1 = run_circuit(da_verifier.clone(), &guest).unwrap();
+
+    // Check that the state transition actually happened
+    assert_eq!(output_1.state_root, [3; 32]);
+    assert!(output_1.unchained_batch_proofs_info.is_empty());
+    assert_eq!(output_1.last_l2_height, 3);
+
+    // Now get more proofs to see the previous light client part is also working correctly
+    let blob_3 = create_mock_blob([3u8; 32], [4u8; 32], 4, true);
+    let blob_4 = create_mock_blob([4u8; 32], [5u8; 32], 5, true);
+
+    let block_header_2 = MockBlockHeader::from_height(2);
+
+    let mock_output_1_serialized = create_prev_lcp_serialized(output_1, true);
+
+    let input_2 = LightClientCircuitInput::<MockDaSpec> {
+        previous_light_client_proof_journal: Some(mock_output_1_serialized),
+        da_block_header: block_header_2,
+        da_data: vec![blob_3, blob_4],
```

Comment:
i thought there could be only 1 blob per mock da block?

Reply:
Yes, but we are not passing da blocks, we are passing blobs

Reply:
and mockda allways verifies?